src/CMakeLists.txt

```diff
@@ -0,0 +1,218 @@
+      "${CMAKE_SOURCE_DIR}/cuda/2d/sart.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/cgls.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/em.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/astra.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/util.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/arith.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/fft.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/2d/darthelper.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/darthelper3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/algo3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/cgls3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/cone_fp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/cone_bp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/fdk.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/par3d_fp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/par3d_bp.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/sirt3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/astra3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/util3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/arith3d.cu"
+      "${CMAKE_SOURCE_DIR}/cuda/3d/mem3d.cu")
+
+  set(ASTRA_CUDA_INCLUDES ${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES})
+
+  # Set CUDA target architectures --------------------------------------------
+
+  if(NOT DEFINED ENV{CUDAARCHS})
+    set(CMAKE_CUDA_ARCHITECTURES "")
+    if((CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 5)
+       AND (CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 11))
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "30;32")
+    endif()
+    if((CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 5)
+       AND (CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 12))
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "35;37")
+    endif()
+    if((CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 6)
+       AND (CMAKE_CUDA_COMPILER_VERSION VERSION_LESS 12))
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "50;52;53")
+    endif()
+    if(CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 8)
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "60;61;62")
+    endif()
+    if(CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 9)
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "70;72")
+    endif()
+    if(CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 10)
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "75")
+    endif()
+    if(CMAKE_CUDA_COMPILER_VERSION VERSION_GREATER_EQUAL 11.1)
+      list(APPEND CMAKE_CUDA_ARCHITECTURES "80;86")
+    endif()
+  endif()
+
+  message(
+    STATUS
+      "CUDA ${CMAKE_CUDA_COMPILER_VERSION} built for archs ${CMAKE_CUDA_ARCHITECTURES}"
+  )
+
+  set_property(TARGET astra PROPERTY CUDA_ARCHITECTURES
+                                     ${CMAKE_CUDA_ARCHITECTURES})
+
+  set_property(
+    TARGET astra
+    APPEND
+    PROPERTY SOURCES ${ASTRA_CUDA_SOURCES})
+
+  set_property(
+    TARGET astra
+    APPEND
+    PROPERTY INCLUDE_DIRECTORIES ${ASTRA_CUDA_INCLUDES})
+
+  # ASTRA_CUDA enables CUDA poritions of code
+  set_property(
+    TARGET astra
+    APPEND
+    PROPERTY COMPILE_DEFINITIONS "ASTRA_CUDA")
+
+  set_property(
+    TARGET astra
+    APPEND
+    PROPERTY LINK_LIBRARIES "cufft")
```

Comment:
This link target is also out of date. We should be using [`find_package(CUDAToolkit)`](https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html) and using the `CUDA::cufft`   and `CUDA::cudart` targets instead of using a plain string.